### PR TITLE
fix!: drop Python 2.7 support

### LINKED ISSUE #228: Drop Python 2.7 support
This would be semver major.

We should verify that clients which cannot move already pin google.resumable-media < 2.0.0dev directly, or transitively.

Extra cleanups:

- ~~Drop `mock` wrapper (use `from unittest import mock` directly).~~ We cannot drop `mock` because Python 3.6 / 3.7 do not have `unittest.mock.AsyncMock` in the stdlib.
- [x] Drop `six`.
- [x] Drop `u"` prefixes for text strings.

### PR DESCRIPTION
**Pull request by tseaver**
- Drop `six`.
- Drop `u"` prefixes for text strings.

Release-As: 2.0.0b1

Closes #228.

### CODE REVIEW

File: README.rst
Comment (plamut) [inline README.rst:23]: Ditto here, AFAIK we should retain the "unsupported Python versions" sections and explain what the last backward-compatible library version is.

File: tests/system/requests/test_upload.py
Comment (plamut) [inline tests/system/requests/test_upload.py:20]: This should be `urllib.parse`.

File: google/_async_resumable_media/__init__.py
Comment (plamut) [inline google/_async_resumable_media/__init__.py:60]: There are also quite a few occurrences of single quoted strings with this `u` prefix.

Also, the script replaced `u` bytes in binary files, i.e. test images, which is probably why a few system tests failed.

File: tests/unit/test__download.py
Comment (plamut) [inline tests/unit/test__download.py:26]: (nit) We can merge this into a single string literal.

**Review by plamut — CHANGES_REQUESTED** [review]
A few nits, but the major thing is the incorrect import. And maybe changes in test image files need to be reverted. Otherwise looks good.

File: tests/system/requests/test_upload.py
Comment (tseaver) [reply tests/system/requests/test_upload.py:20]: e2b99bd5edf634b21b59178843ae75aaffd27408

File: tests/unit/test__download.py
Comment (tseaver) [reply tests/unit/test__download.py]: 3f6d81d

File: README.rst
Comment (tseaver) [reply README.rst:23]: 87c3d72

File: google/_async_resumable_media/__init__.py
Comment (tseaver) [reply google/_async_resumable_media/__init__.py:60]: - `u'` literals: e7cdd35
- Images restored: 9d82941
- Systest import: e2b99bd

**Review by plamut — COMMENTED** [review]
The `six` pin sneaked back into `constraints-3.6.txt` after merging the latest `master`. When that is removed, I think we are good to go.

**Edit:** Ugh, and that `urllib_parse` import again for some reason... (ah, it's actually still there, it was not fixed in all files)

File: tests/system/requests/test_upload.py
Comment (plamut) [reply tests/system/requests/test_upload.py:20]: There's another file where this replacement is needed - `tests/system/requests/test_upload.py`

**Review by plamut — APPROVED** [review]

**Comment by plamut** [comment]
I have nothing more to add, thanks for this cleanup!

**Comment by tseaver** [comment]
Marked as `do not merge` pending discussion over the versioning policy next week with @busunkim96, @crwilcox, et al.

**Pull request merged**
